### PR TITLE
Provide more extensive method documentation.

### PR DESCRIPTION
**Pull request by DJ4ddi**
Something that I feel is sorely missing for this generator is clean XML docs. This PR adds all known properties covered by [NSwag](https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.CodeGeneration.CSharp/Templates/Client.Method.Documentation.liquid) (which is reasonably complete in my opinion).

One extra that I added is response code documentation in the `<throws>` tag. Since this is not a common thing to do, I provided a setting to disable it.

I did not update any generated output (e.g. the files generated by tests or the docfx pages). There is also no command line option to change the setting. Test coverage of the new code is 96%.

### CODE REVIEW

**Comment by christianhelle** [comment]
@DJ4ddi First of all, thanks for taking the time to implement this! 

Great idea and the implementation looks good to me

I'm a bit unsure if Refit at all throws an `ApiException` if the function returns `Task<IApiResponse>` or `Task<IApiResponse<T>>`.

We can merge this in, investigate the actual behavior, then eventually implement checking if the Refitter is configured with the `--use-api-response` and only add the `<exception />` XML docs if exceptions are expected

What do you think @DJ4ddi ?

**Comment by christianhelle** [comment]
@DJ4ddi I pulled your branch and reviewed it locally to do a couple of experiments

I generated 2 Refit clients: using functions that returns `Task<IApiResponse<T>>` and another one that returns `Task<T>`

I ran the code in a small console app

```csharp
using Refit;
using System;
using System.Threading.Tasks;

await TestPetstoreUsingApiResponse();
await TestPetstoreUsingDirectTypes();
await Task.Delay(0);

static async Task TestPetstoreUsingApiResponse()
{
    var responseClient = RestService.For<PetstoreUsingApiResponse.ISwaggerPetstore>("https://petstore3.swagger.io/api/v3");
    var response = await responseClient.GetPetById(1);

    Console.WriteLine("## Using IApiResponse<T> as return type ##");
    Console.WriteLine($"HTTP Status Code: {response.StatusCode}");
    Console.WriteLine($"Name: {response.Content?.Name}");
    Console.WriteLine($"Status: {response.Content?.Status}");

    var pets = await responseClient.FindPetsByStatus(PetstoreUsingApiResponse.Status.Available);
    Console.WriteLine("Found " + pets.Content?.Count + " available pet(s)");

    var taggedPets = await responseClient.FindPetsByTags(new[] { "tag1Updated", "new" });
    Console.WriteLine("Found " + taggedPets.Content?.Count + " tagged pet(s)");
    Console.WriteLine();
}

static async Task TestPetstoreUsingDirectTypes()
{
    var client = RestService.For<Petstore.ISwaggerPetstoreDirect>("https://petstore3.swagger.io/api/v3");
    var pet = await client.GetPetById(1);

    Console.WriteLine("## Using types directly ##");
    Console.WriteLine($"Name: {pet.Name}");
    Console.WriteLine($"Status: {pet.Status}");

    var pets = await client.FindPetsByStatus(Petstore.Status.Available);
    Console.WriteLine("Found " + pets.Count + " available pet(s)");

    var taggedPets = await client.FindPetsByTags(new[] { "tag1Updated", "new" });
    Console.WriteLine("Found " + taggedPets.Count + " tagged pet(s)");
    Console.WriteLine();
}
```

and this is the output I got

```shell
## Using IApiResponse<T> as return type ##
HTTP Status Code: NotFound
Name:
Status:
Found  available pet(s)
Found  tagged pet(s)

## Using types directly ##
Unhandled exception. Refit.ApiException: Response status code does not indicate success: 404 (Not Found).
   at Refit.RequestBuilderImplementation.<>c__DisplayClass14_0`2.<<BuildCancellableTaskFuncForMethod>b__0>d.MoveNext() in /_/Refit/RequestBuilderImplementation.cs:line 288
--- End of stack trace from previous location ---
   at Program.<<Main>$>g__TestPetstoreUsingDirectTypes|0_1() in C:\projects\christianhelle\refitter\test\ConsoleApp\Program.cs:line 33
   at Program.<Main>$(String[] args) in C:\projects\christianhelle\refitter\test\ConsoleApp\Program.cs:line 6
   at Program.<Main>(String[] args)
```

**Comment by christianhelle** [comment]
@all-contributors please add @DJ4ddi for ideas

File: src/Refitter.Core/XmlDocumentationGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        foreach (var parameter in method.Parameters)
+        {
+            if (parameter == null || string.IsNullOrWhiteSpace(parameter.Description))
+                continue;
+
+            this.AppendXmlCommentBlock("param", parameter.Description, code, new Dictionary<string, string>
+                { ["name"] = parameter.VariableName });
+        }
+
+        if (method.HasResult && !string.IsNullOrWhiteSpace(method.ResultDescription))
```
Comment (christianhelle) [inline src/Refitter.Core/XmlDocumentationGenerator.cs]: Changing

```csharp
        if (method.HasResult && !string.IsNullOrWhiteSpace(method.ResultDescription))
        {
            this.AppendXmlCommentBlock("returns", method.ResultDescription, code);
        }

        this.AppendXmlCommentBlock("throws", this.BuildErrorDescription(method.Responses), code, new Dictionary<string, string>
            { ["cref"] = "ApiException" });
```

into

```csharp
        if (_settings.ReturnIApiResponse)
        {
            this.AppendXmlCommentBlock("returns", "An instance of IApiResponse", code);
        }
        else
        {
            if (method.HasResult && !string.IsNullOrWhiteSpace(method.ResultDescription))
            {
                this.AppendXmlCommentBlock("returns", method.ResultDescription, code);
            }
        }

        if (!_settings.ReturnIApiResponse)
            this.AppendXmlCommentBlock("throws", this.BuildErrorDescription(method.Responses), code, new Dictionary<string, string>
                { ["cref"] = "ApiException" });
```

will generate the `<throws />` XML docs when the success response type is directly returned

```csharp
        /// <summary>Add a new pet to the store</summary>
        /// <remarks>Add a new pet to the store</remarks>
        /// <param name="body">Create a new pet in the store</param>
        /// <returns>Successful operation</returns>
        /// <throws cref="ApiException">
        /// Thrown when the request returns a non-success status code:
        /// 405: Invalid input
        /// </throws>
        [Headers("Accept: application/xml, application/json")]
        [Post("/pet")]
        Task<Pet> AddPet([Body] Pet body);
```

and when the return type is `IApiResponse` or `IApiResponse<T>` then the generated comments will look like this:

```csharp
        /// <summary>Add a new pet to the store</summary>
        /// <remarks>Add a new pet to the store</remarks>
        /// <param name="body">Create a new pet in the store</param>
        /// <returns>An instance of IApiResponse</returns>
        [Headers("Accept: application/xml, application/json")]
        [Post("/pet")]
        Task<IApiResponse<Pet>> AddPet([Body] Pet body);
```

Comment (DJ4ddi) [reply src/Refitter.Core/XmlDocumentationGenerator.cs]: You are correct, I did not consider the `IApiResponse` case. I'll make changes as suggested with one addition: I'd like to follow the same status code pattern as in the exception (meaning that if a generic response is returned, specific codes and their description can be added to the documentation).

Comment (christianhelle) [reply src/Refitter.Core/XmlDocumentationGenerator.cs]: That would be great @DJ4ddi

**Review by christianhelle — APPROVED** [review]
Excellent work @DJ4ddi

**Pull request merged**